sdk/download.go

```diff
@@ -318,3 +319,18 @@ func UpdateSignedFile(file, url string, client *http.Client) error {
 	plog.Infof("Verified file: %s", file)
 	return nil
 }
+
+func DetectSDKVersion(url, branch string) (string, error) {
```

Comment:
Yes, but not sure what I want that behaviour to be. Maybe replace could start skipping the repo init/sync if it isn't needed, or at least make it optional but I think picking that behavior is orthogonal to this.